source/merge-deep.d.ts

```diff
@@ -76,9 +79,12 @@ type Rest5 = PickRestType<string[]>; // => string[]
 ```
 */
 type PickRestType<Type extends UnknownArrayOrTuple> = number extends Type['length']
-	? ArrayTail<Type> extends [] ? Type : PickRestType<ArrayTail<Type>>
+	? PickRestType___<ArrayTail<Type>, Type>
 	: [];
 
+// Utility to avoid computing ArrayTail twice.
+type OmitRestType___<Tail extends UnknownArrayOrTuple, Type extends UnknownArrayOrTuple, Result extends UnknownArrayOrTuple = []> = Tail extends [] ? Result : OmitRestType<Tail, [...Result, FirstArrayElement<Type>]>;
```

Comment:
Can we maybe break this up over a few more lines – its hard for the eyes to follow the flow I think

Reply:
Done ;)